src/main/java/app/controller/settings/SettingsObject.java

```diff
@@ -18,16 +21,35 @@ public SettingsObject(Rectangle2D rectangle, FurnitureType type)
                 rectangle.getHeight());
         this.type = type;
         this.teleportTo = null;
+        this.teleportRotation = null;
+        this.textureType = null;
+        this.textureOrientation = null;
     }
 
-    public SettingsObject(Rectangle2D rectangle, Vector teleportTo)
+    public SettingsObject(Rectangle2D rectangle, Vector teleportTo, double teleportRotation)
```

Comment:
Why are you adding this?!  We don't have the functionality to support this yet, so you are adding more complexity, more code and overloading constructors multiple times without adding any features! 

YAGNI  -You aren't gonna need it